htcanalyze/main.py

```diff
@@ -44,18 +51,18 @@
                          "errors", "ram-history"]
 
 
-# class to store and change global variables
-class GlobalPlayer(object):
+# this class sets the output and input streams as global variables
+class GlobalStreams(object):
```

Comment:
This class is redundant. The only relevant parts in it are `sys.stdin.isatty()`, `sys.stdin.readlines()`, and `sys.stdout.isatty()`, which you can call directly rather than assign them to class variables.

Of course, if you intend to expand upon this class later by offering your own options that can overwrite the `sys` output, then this class becomes relevant. But if you intend to do so, please add a `# TODO:`, so you don't forget and we know this code is not redundant.

Reply:
As I commented earlier:
> Hm actually not that easy, because sys.stdin and sys.stdout can change at runtime, because of for ex. the ram-histogram created by plotille and I think even by the progress bars of rich.

I admit, that this class is really confusing and I was sure I've adding this peace of knowledge somewhere. 
However some of the tools I am using change the behaviour of sys.stdin and sys.stdout at runtime.
I try to catch the state directly after starting the script.
If you have a solution for that behaviour or any advice, how to prevent this, please let me know, because I'm standing in front of this Issue for quite a while and this was my way of solving it.

Reply:
But actually if I just catch this in the run method, I might get around using this class.
In earlier releases I was using this segment quite more often. I have reduced the use, so that it should actually be easier to get rid of this class now. I will have an eye on that, but I am still a bit pessimistic about it.

Reply:
Seeing how you use the class only once, you may as well assign the `sys` values to variables in `main()` or `run()` (since you only request them in `run()`), and then pass them along to the relevant location(s).